### DISCUSSION TITLE
maplibre L7 Plugin

### DISCUSSION THREAD

**Discussion by Carlos-Carreno-Berlanga** [maplibre/maplibre-gl-js#1522] [Q&A]
Hi I am a big fan of this project,

Keep the good work please.

 But I have one question, I've read in this link https://maplibre.org/maplibre-gl-js-docs/plugins/ that we can use Maplibre with L7 ( https://github.com/antvis/L7 ) but I haven't been able to figure it out how. 

My understanding is that L7 has a dependency on Mapbox, and it will be easy to add another dependency to Maplibre, but that is not the case at the moment. 

Please correct me if I am wrong, maybe I am missing something.

Thanks in advance.

**Comment by wipfli** [comment]
I took the list of plugins from the Mapbox documentation without actually checking that all the plugins work with MapLibre. Maybe this one does not work with MapLibre?

**Comment by wipfli** [comment]
Though the string "mapbox" does not seem to be in https://github.com/antvis/L7/blob/master/package.json... Anyway, if you should find out that L7 is incompatible with MapLibre, feel free to ask the maintainers of L7 to add MapLibre support and feel also free to update the docs website with a pull request. The relevant file for the plugins is https://github.com/maplibre/maplibre-gl-js-docs/blob/main/docs/data/plugins.json

**Comment by junhaotong** [comment]
Maplibre and Mapbox have the same api. So you can use like this：
```
import { Map } from '@antv/l7-maps';
import { Scene } from '@antv/l7';
.......

const scene = new Scene({
          id: 'container',
          map: new Map({
            mapInstance: mapView,   // Maplibre Instance
          }),
        });
```
 Then L7 can work with Maplibre now.
